auto-remediate/security_hub_rules.py

```diff
@@ -67,58 +67,62 @@ def iam_user_unused_credentials_check(self, resource_id):
         client = boto3.client('iam')
         
         try:
-            response = client.list_users()
+            paginator = client.get_paginator('list_users')
         except:
-            self.logging.error("Could not list all IAM users.")
+            self.logging.error("Could not get a paginator to list all IAM users.")
             self.logging.error(sys.exc_info()[1])
             return False
         
-        for user in response.get('Users'):
-            if resource_id == user.get('UserId'):
-                user_name = user.get('UserName')
-                
-                # check password usage
-                try:
-                    login_profile = client.get_login_profile(UserName=user_name)
-                    login_profile_date = login_profile.get('LoginProfile').get('CreateDate')
-                except:
-                    self.logging.error(f"Could not retrieve IAM Login Profile for User '{user_name}'.")
-                    self.logging.error(sys.exc_info()[1])
-                    return False
-                
-                if SecurityHubRules.get_day_delta(login_profile_date) > 90:
+        for response in paginator.paginate(PaginationConfig={'PageSize': 100}):
```

Comment:
we should stop iterating the paginator.paginate() once user is found
so maybe like
https://stackoverflow.com/questions/2361426/get-the-first-item-from-an-iterable-that-matches-a-condition

```python
user = None
while not user:
    for response in paginate ...
        user = next((u for u in response['Users'] if u['UserId'] == resource_id), None)
         # do remediation for user 
        break
```

Reply:
@jihygk I've got a `return True` within the if once the remediation is done. Will that not finish the function early?

Reply:
yeah let's not worry about it for now, with 100 users/page i think we should be ok except extreme cases